dspace-server-webapp/src/test/java/org/dspace/app/rest/GroupRestRepositoryIT.java

```diff
@@ -1538,4 +1650,342 @@ public void removeMemberUnprocessableTest() throws Exception {
+            community = communityService.create(null, context);
+            collection = collectionService.create(context, community);
+            adminGroup = collectionService.createAdministrators(context, collection);
+            worfklowGroup = collectionService.createWorkflowGroup(context, collection, 1);
+            submitterGroup = collectionService.createSubmitters(context, collection);
+            context.commit();
+            adminGroup = context.reloadEntity(adminGroup);
+
+            String authToken = getAuthToken(admin.getEmail(), password);
+            getClient(authToken).perform(
+                    get("/api/eperson/groups/" + adminGroup.getID() + "/object")
+            ).andExpect(status().isOk());
+            getClient(authToken).perform(
+                    get("/api/eperson/groups/" + worfklowGroup.getID() + "/object")
+            ).andExpect(status().isOk());
+            getClient(authToken).perform(
+                    get("/api/eperson/groups/" + submitterGroup.getID() + "/object")
+            ).andExpect(status().isOk());
+        } finally {
+            if (collection != null) {
+                CollectionBuilder.deleteCollection(collection.getID());
+            }
+            if (community != null) {
+                CommunityBuilder.deleteCommunity(community.getID());
+            }
+            if (adminGroup != null) {
+                GroupBuilder.deleteGroup(adminGroup.getID());
+            }
+            if (worfklowGroup != null) {
+                GroupBuilder.deleteGroup(worfklowGroup.getID());
+            }
+            if (submitterGroup != null) {
+                GroupBuilder.deleteGroup(submitterGroup.getID());
+            }
+        }
+    }
+
+    @Test
+    public void getGroupObjectNotFoundTest() throws Exception {
+        GroupService groupService = EPersonServiceFactory.getInstance().getGroupService();
+        Group adminGroup = null;
+        try {
+            context.turnOffAuthorisationSystem();
+            adminGroup = groupService.create(context);
+            context.commit();
+            adminGroup = context.reloadEntity(adminGroup);
+
+            String authToken = getAuthToken(admin.getEmail(), password);
+            getClient(authToken).perform(
+                    get("/api/eperson/groups/" + adminGroup.getID() + "/object")
+            ).andExpect(status().isNotFound());
+        } finally {
+            if (adminGroup != null) {
+                GroupBuilder.deleteGroup(adminGroup.getID());
+            }
+        }
+    }
+
+    @Test
+    public void getGroupObjectUnauthorizedTest() throws Exception {
+        GroupService groupService = EPersonServiceFactory.getInstance().getGroupService();
+        Group adminGroup = null;
+        try {
+            context.turnOffAuthorisationSystem();
+            adminGroup = groupService.create(context);
+            context.commit();
+            adminGroup = context.reloadEntity(adminGroup);
```

Comment:
Same as above. should use GroupBuilder, missing a context.restoreAuthSystemState(); and same question about commit() and reloadEntity() as above.